os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -191,6 +191,13 @@ void binary_manager_add_binlist(FAR struct tcb_s *tcb)
 {
```

Comment:
NULL checking is right but a scheduling is locked by sched_lock before calling task_activate in pthread_create. Is it possible that pthread is exited before calling it?

Reply:
Yes. But as per my understanding this is what happens:
Both parent and child tasks are created with default priority of 100.
- Parent task is running
- pthread_create
- activate task
- pthread_sem_take(&pjoin->data_sem, false); is called ... So parent gets blocked
- pthread_start for child is called
- Here, pthread_sem_give(&pjoin->data_sem) is called. This is supposed to unblock the parent. But since child is the current running task and both have same priority, the parent task gets added to the ready to run list just after the child task.
- So, the child task will continue to run, before the pthread_create of the parent is completed.

This is why, i think, its possible that the child pthread gets exited before the pthread_create is complete.

Reply:
Thank you so much. I understood